### PR TITLE
Convert `src/SecurityManager.js` to TypeScript

### PR DESCRIPTION
**Pull request by jryans**
This includes a few small API tweaks as well, only in cases where TS revealed we
were doing something confusing or wrong.

Part of https://github.com/vector-im/element-web/issues/15350
Depends on https://github.com/matrix-org/matrix-js-sdk/pull/1502

### CODE REVIEW

**Comment by jryans** [comment]
CI is failing because those tasks don't attempt to pull matching branches of JS SDK. 😒 I'll work on updating the pipelines.

File: src/SecurityManager.ts [hunk 1]
```diff
@@ -72,7 +77,9 @@ async function confirmToDismiss() {
     return !sure;
 }
 
-function makeInputToKey(keyInfo) {
+function makeInputToKey(
+    keyInfo: ISecretStorageKeyInfo,
+): ({ passphrase, recoveryKey }: { passphrase: string, recoveryKey: string }) => Promise<Uint8Array> {
```
Comment (turt2live) [inline src/SecurityManager.ts:82]: Should be able to name this without freaking out the compiler
```suggestion
): (keyParams: { passphrase: string, recoveryKey: string }) => Promise<Uint8Array> {
```

**Review by turt2live — APPROVED** [review]
seems generally sane

File: src/SecurityManager.ts (hunk 1 above)
Comment (jryans) [reply src/SecurityManager.ts:82]: Aha, indeed... Is there generally a "good" way to type object destructuring, as in [`getSecretStorageKey`](https://github.com/matrix-org/matrix-react-sdk/pull/5307/files/2698a12a6fc61e8e9a1d992b7054f116d355651f#diff-e94b1d35af05f12c5eaf5630b5d4fe9dR97)...? It's quite verbose at the moment, but I think both sides are needs for the destructuring and typing to both take effect...?

**Pull request merged**